script.js

```diff
@@ -0,0 +1,306 @@
+        face: '2',
+        value: 2
+    },
+    {
+        face: '3',
+        value: 3
+    },
+    {
+        face: '4',
+        value: 4
+    },
+    {
+        face: '5',
+        value: 5
+    },
+    {
+        face: '6',
+        value: 6
+    },
+    {
+        face: '7',
+        value: 7
+    },
+    {
+        face: '8',
+        value: 8
+    },
+    {
+        face: '9',
+        value: 9
+    },
+    {
+        face: '10',
+        value: 10
+    },
+    {
+        face: 'J',
+        value: 11
+    },
+    {
+        face: 'Q',
+        value: 12
+    },
+    {
+        face: 'K',
+        value: 13
+    }
+]
+
+const slider = document.getElementById("difficulty")
+const startBtn = document.getElementById("start")
+let side = 'R';
+let difficulty = 0
+let cards = [];
+let drawnCards = [];
+let matches = 0;
+let yourMatches = 0;
+let yourLosses = 0;
+
+function createDeck() {
+    suits.forEach(s => {
+        faces.forEach(f => {
+            let card = {
+                suit: s.suit,
+                color: s.color,
+                face: f.face,
+                value: f.value
+            }
+            cards.push(card);
+        })
+    });
+}
+
+function drawCard(cards) {
+    let rand = Math.floor(Math.random() * cards.length)
+    let chosenCard =  cards.splice(rand, 1)[0]
+    drawnCards.push(chosenCard);
+    return chosenCard
+}
+
+function startTheClock(difficulty) {
+    let count = cards.length;
+    moveCards(drawCard(cards))
+
+    const timer = setInterval(function() {
+        count--;
+        if (count == 0) {
+            clearInterval(timer);
+            console.log("No more cards!");
+            console.log(`${yourMatches} out of ${matches} matches`)
+            let message = document.querySelector("#message>p")
+            message.innerHTML = `You got<br>${yourMatches}<br>out of<br>${matches}<br>matches`
+        } else {
+            moveCards(drawCard(cards));
+            if(sameNumber() || oneUpOneDown() || equalToTen()) {
+                matches++
+            }
+        }
+    }, difficulty);
+}
+
+function sameNumber() {
+    let currentCard = drawnCards[drawnCards.length - 1]
+    let previousCard = drawnCards[drawnCards.length - 2]
+    if(currentCard.value == previousCard.value) {
+        console.log('match')
+        return true;
+    }
+}
+
+function oneUpOneDown() {
+    let currentCard = drawnCards[drawnCards.length - 1]
+    let previousCard = drawnCards[drawnCards.length - 2]
+    if
+    (
+        currentCard.value - 1 == previousCard.value ||
+        currentCard.value == previousCard.value ||
+        currentCard.value + 1 == previousCard.value
+    )
+    {
+        console.log('match')
+        return true;
+    }
+}
+
+function equalToTen() {
+    let currentCard = drawnCards[drawnCards.length - 1]
+    let previousCard = drawnCards[drawnCards.length - 2]
+    if(currentCard.value + previousCard.value == 10) {
+        console.log('match')
+        return true;
+    }
+}
```

Comment:
You should rewrite these functions so that they do not rely on lexical scope / side effects.

Just pass in two cards.
```suggestion
function sameNumber(card1, card2) {
    if(card1.value == card2.value) {
        return true;
    } else {
        return false;
    }
}
etc.
```